Improve Documentation and VSCode Support

Xeratec opened the pull request:
This PR introduces multi-branch GitHub Pages support to avoid overwriting existing documentation with each push, and improves the integration with Visual Studio Code to simplify debugging workflows. It also extends the documentation with a tutorial on using the debug print topology pass and the introspective code transformation (see #75).

Furthermore, this PR cleans up the compilation flow for PULP and Siracusa by disabling verbose build output by default. Users can still increase verbosity using the `-v` flag, for example:
```sh
python testRunner_siracusa.py -t Tests/Adder  -vvv
```

### VS Code Support
To streamline development in Visual Studio Code, the build system now creates symlinks from the most recent build directory and generated source files (e.g., generated C files and `deployStates`) into a unified `TEST_RECENT` folder. This enables IntelliSense and other language features by pointing VSCode to the generated `compile_commands.json,` facilitating easier navigation and debugging.

### Multi Branch Support
The workflow supports multi-branch GitHub Pages, enabling documentation for different branches to be deployed side-by-side. This allows maintaining documentation for various development stages (e.g., `main`, feature branches, and PR-specific previews). Each branch’s documentation is hosted under its respective path. For example:
- The default branch (`main`) is deployed to the root of the GitHub Pages site.
  (e.g. https://pulp-platform.github.io/Deeploy/)
- Other branches, such as feature branches, are deployed under sub-paths corresponding to their branch names.
  (e.g. https://pulp-platform.github.io/Deeploy/branch/devel)

An example documentation for this branch can be previewed [here](https://xeratec.github.io/Deeploy/branch/pr--improve_debugging/index.html).

## Added
- Tutorial section in the documentation
- Guide on using the debug print topology pass and code transformation
- VSCode configuration files for improved IDE support
-Multi-branch GitHub Pages deployment support

## Changed
- Updated `README.md` description to use a persistent development container
- Symlinking of the latest build and source files into `TEST_RECENT`
- Disabled `CMAKE_VERBOSE_MAKEFILE` by default for cleaner builds.

## Fixed
- Removed unsupported `-MMD` compiler flag in LLVM-based toolchains.

## PR Merge Checklist

1. [x] The PR is rebased on the latest `devel` commit and pointing to `devel`.
2. [x] Your PR reviewed and approved.
3. [x] All checks are passing.
4. [x] The `CHANGELOG.md` file has been updated.
5. [x] If the docker was modified, change back its link after review.
6. [x] Merge #75 and rebase

Victor-Jung commented on .vscode/launch.json:
```diff
@@ -0,0 +1,105 @@
+            "id": "additionalArgsUntiled",
+            "type": "promptString",
+            "description": "Additional Arguments",
+            "default": ""
```
Fixed it [here](https://github.com/Victor-Jung/Deeploy/commit/c2de0d1a5b483face3f73cd55a5d2b4bea8b328d) by putting -v as default arg and cleaning the JSON syntax

Xeratec replied:
Adapted here [5a01f40](https://github.com/pulp-platform/Deeploy/pull/76/commits/5a01f40d24cd3eaf97462951c5d9f505c4c5e2a5). However, I kept the `additionalArgsUntiled` and `additionalArgsTiled` separate as I want `--doublebuffer` for tiled architectures by default.

Xeratec commented on docs/tutorials/debugging.rst:
```diff
@@ -0,0 +1,59 @@
+        MemoryAwarePrintOutputGeneration("L1")
+    ])
+
+To apply these code transformations across all bindings, refer to the test implementation in ``DeeployTest/testPrintInputOutputTransformation.py``.
```
Adapted here [5a01f40](https://github.com/pulp-platform/Deeploy/pull/76/commits/5a01f40d24cd3eaf97462951c5d9f505c4c5e2a5).

Xeratec commented on .vscode/launch.json:
```diff
@@ -0,0 +1,105 @@
+                "MemPool",
+                "Apollo3",
+                "Apollo4",
+                "Snitch",
```
Adapted here [5a01f40](https://github.com/pulp-platform/Deeploy/pull/76/commits/5a01f40d24cd3eaf97462951c5d9f505c4c5e2a5).

Xeratec commented:
Also fixed a typo and added the `--platform` flag, which is required to use the container on Apple Silicon. In the future, I suggest we look into building an ARM container. See [3d455b3](https://github.com/pulp-platform/Deeploy/pull/76/commits/3d455b3fa4c6a18cc6757ec693873c3b8652fd84)

Victor-Jung reviewed (approved):
LGTM. Can you open an issue for the ARM container?

Xeratec commented:
> LGTM. Can you open an issue for the ARM container?

Done in #77.
